core/src/context.rs

```diff
@@ -425,29 +449,42 @@ where
             )
         };
 
-        let wasi_env = if root.is_none() || !(read || write || create) {
-            WasiState::new("")
+        let has_fs_access = root.is_some() && (read || write || create);
+
+        let wasi_env = {
+            // Get all the variables that this element has read access to
+            let vars_to_read: Vec<(String, String)> = self
+                .get_vars_to_read(from, output_format)?
+                .into_iter()
+                .filter_map(|(name, ty)| {
+                    self.state
+                        .variables
+                        .get(&name)
+                        .filter(|value| value.get_type() == ty)
```

Comment:
We talked about this in person so I'll mark it as resolved. I think it should be pretty reasonable decision. But, yeah it can be pretty confusing how we should deal about potential type mismatches. I guess the risk of doing it this ways is that users may use the wrong type and not get a error about doing so. But in most cases where a variable is "required" the package can create a warning or error about "variable ____ is not defined"